tools/install_mac_deps.sh

```diff
@@ -5,30 +5,38 @@
 set -e  # Exit on error.
 set -x  # Make command execution verbose
 
-# Install some custom requirements on OS X using brew
-BREW=/usr/local/bin/brew
-if [[ ! -x $BREW ]]; then
-    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
-fi
 
-BOOST_LIB="boost@1.76"
+# Set up Homebrew differently for arm64.
+if [[ $(uname -m) == 'arm64' ]]; then
+    (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> ~/.zprofile
```

Comment:
Is it a clean environment each time CI runs? I assume it is, but asking in case we need to check for the eval line before adding it to .zprofile.

Reply:
Do you have a suggestion for a cleaner command to set up `brew`? I do not have a Mac so I am just pasting what people suggest.

Reply:
Ideally, one should be able to run this command locally, too.

Reply:
This seems reasonable to me.

Reply:
Used ChatGPT to "enhance" the script with some guards, please take another look.